codexctl/__init__.py

```
@@ -303,7 +305,30 @@ def call_func(self, function: str, args: dict) -> None:
 
                 if device_version_uses_new_engine:
-                    remarkable.install_sw_update(update_file)
+                    if not manual_dd_update:
+                        remarkable.install_sw_update(update_file)
+                    else:
+                        try:
+                            from .analysis import get_update_image
+                        except ImportError:
+                            raise ImportError(
+                                "remarkable_update_image is required for this update. Please install it!"
+                            )
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)